compiler/dyno/test/resolution/testTypeConstruction.cpp

```diff
@@ -1231,7 +1231,36 @@ static void test39() {
   assert(fields->fieldType(0).type() == ct);
 }
 
+static void test40() {
```

Comment:
Unless I'm not seeing it in the other existing tests, it might be good to try to cover the other cases, which I think are something that results in unknown type and something that doesn't have substitutions.

Reply:
The UnknownType should only be used during resolution so the testing strategy here shouldn't be able to find it.. We have other tests that do type construction with concrete types (and also other tests that check functions with concrete types). I will add a variant of this test that works with a concrete type and a variant that has `R(1)` at the call site.

Reply:
> ... so the testing strategy here shouldn't be able to find it...

Is it able to be found/exercised by another existing testing strategy then?

Reply:
Yes but I don't want to get in to that right now for this PR. Happy to put a note for Future Work though if you like.

Reply:
Yes please